integration/stacks_test.go

```diff
@@ -140,15 +143,123 @@ func deployStack(ctx context.Context, oktetoPath, stackPath string) error {
 	cmd.Env = os.Environ()
-	cmd.Dir = stackGitFolder
+	cmd.Dir = dir
 	o, err := cmd.CombinedOutput()
 	if err != nil {
 		return fmt.Errorf("okteto stack destroy failed: %s - %s", string(o), err)
 	}
 	log.Printf("okteto stack destroy success")
 	return nil
 }
+
+func TestCompose(t *testing.T) {
+	if mode == "client" {
+		t.Skip("this test is not required for client-side translation")
+		return
+	}
+
+	ctx := context.Background()
+	oktetoPath, err := getOktetoPath(ctx)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	tName := fmt.Sprintf("TestStacks-%s", runtime.GOOS)
+	name := strings.ToLower(fmt.Sprintf("%s-%d", tName, time.Now().Unix()))
+	namespace := fmt.Sprintf("%s-%s", name, user)
+
+	t.Run(tName, func(t *testing.T) {
+		log.Printf("running %s \n", tName)
+		k8Client.Reset()
+	})
+	if err := createNamespace(ctx, oktetoPath, namespace); err != nil {
+		t.Fatal(err)
+	}
+
+	log.Printf("created namespace %s \n", namespace)
+
+	if err := cloneGitRepo(ctx, composeGitRepo); err != nil {
+		t.Fatal(err)
+	}
+
+	if err := deployStack(ctx, oktetoPath, "docker-compose.yml", composeGitFolder); err != nil {
+		t.Fatal(err)
+	}
+	defer deleteGitRepo(ctx, composeGitFolder)
+
+	log.Printf("deployed stack using %s \n", "docker-compose.yml")
+
+	endpoint := fmt.Sprintf("https://nginx-%s.cloud.okteto.net/db", namespace)
+	content, err := getContent(endpoint, 150)
+	if err != nil {
+		t.Fatalf("failed to get stack content: %s", err)
+	}
+	var objmap map[string]json.RawMessage
+	err = json.Unmarshal([]byte(content), &objmap)
+
+	if strings.ToUpper(string(objmap["lower"])) != string(objmap["uppercase"]) {
+		t.Fatal("Not the right response")
+	}
+
+	if err := destroyStack(ctx, oktetoPath, "docker-compose.yml", composeGitFolder); err != nil {
+		t.Fatal(err)
+	}
+
+	log.Println("destroyed stack")
+
+	time.Sleep(5 * time.Second)
+	_, err = getDeployment(ctx, namespace, "api")
+	if err == nil {
+		t.Fatalf("'api' deployment not deleted after 'okteto stack destroy'")
+	}
+	if !strings.Contains(err.Error(), "not found") {
+		t.Fatalf("error getting deployment 'api': %s", err.Error())
+	}
+	_, err = getDeployment(ctx, namespace, "producer")
+	if err == nil {
+		t.Fatalf("'producer' deployment not deleted after 'okteto stack destroy'")
+	}
+	if !strings.Contains(err.Error(), "not found") {
+		t.Fatalf("error getting deployment 'producer': %s", err.Error())
+	}
+	_, err = getDeployment(ctx, namespace, "consumer")
+	if err == nil {
+		t.Fatalf("'consumer' deployment not deleted after 'okteto stack destroy'")
+	}
+	if !strings.Contains(err.Error(), "not found") {
+		t.Fatalf("error getting deployment 'consumer': %s", err.Error())
+	}
+	_, err = getDeployment(ctx, namespace, "web-svc")
+	if err == nil {
+		t.Fatalf("'web-svc' deployment not deleted after 'okteto stack destroy'")
+	}
+	if !strings.Contains(err.Error(), "not found") {
+		t.Fatalf("error getting deployment 'web-svc': %s", err.Error())
+	}
+
+	_, err = getStatefulset(ctx, namespace, "mongodb")
```

Comment:
would it make sense to also check that the volume is destroyed?

Reply:
The destroy command doesn't destroy volumes by default. You have to specify it by adding `-v` flag